client/runtime.go

```diff
@@ -37,9 +41,26 @@ import (
 
 // TLSClientOptions to configure client authentication with mutual TLS
 type TLSClientOptions struct {
-	Certificate        string
-	Key                string
-	CA                 string
+	// Certificate is the path to a PEM-encoded certificate to be used for
+	// client authentication. If set then Key must also be set.
+	Certificate string
+
+	// LoadedCertificate is the certificate to be used for client authentication.
+	// This field is ignored if Certificate is set. If this field is set, LoadedKey
+	// is also required.
+	LoadedCertificate *x509.Certificate
+
+	// Key is the path to an unencrypted PEM-encoded private key for client
+	// authentication. This field is required if Certificate is set.
+	Key string
+
+	// LoadedKey is the key for client authentication. This field is required if
+	// LoadedCertificate is set.
+	LoadedKey crypto.PrivateKey
+
+	// CA is a path to a PEM-encoded certificate.
```

Comment:
Since you're already expanding the docs, perhaps this should explain what CA is used for, that it's the intermediate certificate(s) for the Certificate Authority that issued the Certificate.